utility/pbkdf2-hash-generator/src/main/java/org/wso2/carbon/identity/utility/hash/generator/pbkdf2/PBKDF2HashGenerator.java

```
@@ -0,0 +1,128 @@
+            String originalPassword = scanner.nextLine();
+
+            String generatedSalt = generateSaltValue();
+            byte[] generatedSecuredPasswordHash;
+            try {
+                generatedSecuredPasswordHash = calculateHash(originalPassword.toCharArray(), generatedSalt);
+            } catch (HashProviderException e) {
+                log.error(e);
+                break;
+            }
+            String strongPassword = bytesToBase64(generatedSecuredPasswordHash);
+
+            log.info("Hashed Password:" + strongPassword);
+            log.info("Salt Value:" + generatedSalt);
+            log.info("----------------------------------");
+        }
+    }
+
+    /**
+     * This method used to initialize the PBKDF2HashProvider instance.
+     */
+    private static void initHashProvider() {
+
+        pbkdf2HashProvider = new PBKDF2HashProvider();
+        pbkdf2HashProvider.init();
+    }
+
+    /**
+     * Generate hash password for the given plain password and salt value.
+     *
+     * @param plainText The plain text value to be hashed.
+     * @param salt      The salt value.
+     * @return The resulting hash value of the value.
+     */
+    private static byte[] calculateHash(char[] plainText, String salt) throws HashProviderException {
+
+        return pbkdf2HashProvider.calculateHash(plainText, salt);
+    }
+
+    /**
+     * This private method returns a saltValue using SecureRandom.
+     *
+     * @return saltValue.
+     */
+    private static String generateSaltValue() throws RuntimeException {
+
+        try {
+            SecureRandom secureRandom = SecureRandom.getInstance(SHA_1_PRNG);
+            byte[] bytes = new byte[16];
+            //secureRandom is automatically seeded by calling nextBytes
+            secureRandom.nextBytes(bytes);
+            return bytesToBase64(bytes);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("SHA1PRNG algorithm could not be found.");
```

its not recommended to throw RuntimeExceptions unless necessary. Shall we throw a checked exception from here and handle it properly where this method is getting called.

---

Addressed with https://github.com/wso2-extensions/identity-hash-provider-pbkdf2/pull/12